pytensor/tensor/math.py

```diff
@@ -2937,7 +2937,58 @@ def matmul(x1: "ArrayLike", x2: "ArrayLike", dtype: Optional["DTypeLike"] = None
     return out
 
 
+def nan_to_num(x, nan=0.0, posinf=None, neginf=None):
+    """
+    Replace NaN values with the `nan` keyword, +INF with the `posinf`
+    keyword and -INF with the `neginf` keyword.  By default NaN will
+    be replaced with 0, +INF with the largest value representable by
+    ``x.dtype`` and -INF with the smallest value representable by
+    ``x.dtype``.  Mimics numpy.nan_to_num for the pytensor format
+    (with the exception that in-place operations are not supported).
+
+    Parameters
+    ----------
+    x
+        Input array.
+    nan
+        The value to replace NaN's with in the tensor (default = 0).
+    posinf
+        The value to replace +INF with in the tensor (default max
+        in range representable by ``x.dtype``).
+    neginf
+        The value to replace -INF with in the tensor (default min
+        in range representable by ``x.dtype``).
+
+    Returns
+    -------
+    out
+        The tensor with NaN's, +INF, and -INF replaced with the
+        specified and/or default substitutions.
+    """
+
+    # Replace NaN's with nan keyword
+    x = switch(isnan(x), nan, x)
```

Comment:
Instead of overriding `x`, let's write all the checks `isnan`\`isposinf`\`isneginf` on the original array.